Update scaladoc docs (#12787)
lampepfl/dotty: docs/docs/usage/scaladoc/changing-versions.md

---

> title: Multiple versions changing

Reviewer: ```suggestion
title: Site versioning
```

Reviewer 2: Maybe change the file name as well?

---

# {{ page.title }}

> Scaladoc now provides convenient way to link between different versions of documentation. The feature is pretty handy if we want to expose older docs for users that didn't migrate to the new version of our library.

Reviewer: ```suggestion
Scaladoc provides a convenient way to switch between different versions of the documentation. The feature is useful if we want to expose older docs for users that didn't migrate to the new version of our library.
```

### How to setup it

> The feature was designed for easy scalability with no need to regenerate all scaladocs after adding next version. Becuase of that we decided to introduce the new setting passed during creation of scaladoc `-versions-dictionary-url` pointing with URL to JSON document holding information about locations of specific versions. The JSON file has single property "versions" that holds dictionary of labels of specific docs and URL pointing to their index.html top-level file.

Reviewer: ```suggestion
The feature was designed for easy scalability with no need to regenerate all scaladocs after adding a new version. To do so a new setting is introduced:  `-versions-dictionary-url`. Its argument must be a URL to a JSON document holding information about the locations of specific versions. The JSON file has single property `versions` that holds the dictionary associating the labels of specific versions of the documentation to the URLs pointing to their index.html
```

Example JSON file:
```
{
  "versions": {

>     "3.0.X": "https://dotty.epfl.ch/3.0.X/docs/index.html",

Reviewer 2: Not very important, but I think it is more common to use lower case here:

```suggestion
    "3.0.x": "https://dotty.epfl.ch/3.0.x/docs/index.html",
```

    "Nightly": "https://dotty.epfl.ch/docs/index.html"
  }
}
```

> This enforce us to provide the setting while genereting docs for each of the versions, however it gives us more flexebility later. Consider one want to add next version of the API docs along previous 5 versions he/she has already published. The only thing is to upload them to some web server host and update the JSON file with another entry. All scaladocs will now become aware of the new scaladoc version.

Reviewer: ```suggestion
This enforce us to provide the setting while genereting docs for each of the versions, however it gives us more flexebility later. 
If you want to add a version of the API docs next to the previous 5 versions that you have already published, then you only need to upload the new docs to a web server and add a new entry to the JSON file. All versions of the site will now become aware of the new site version.
```

### How does it look from user perspective

> Providing JSON file via `-versions-dictionary-url` enables scaladoc to link between versions. We found it convenient to change the revision label into drop-down menu that let you change between different version. Everything will change automatically, no need for further setup.

Reviewer: ```suggestion
Providing a JSON file via `-versions-dictionary-url` enables scaladoc to link between versions. It is also convenient to be able to change the revision label in the drop-down menu. Everything will change automatically.
```
